lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -78,7 +78,7 @@ def self.deliver(*args)
       automate_attrs = automate_attrs_from_options(options)
 
       if options[:object_type]
-        vmdb_object = options[:object_type].constantize.find_by(:id => options[:object_id])
+        vmdb_object = options[:object_type].constantize.find_by!(:id => options[:object_id])
```

Comment:
@pkomanek 
Can't this just be tested by creating 
```
  options = {}
 options[:object_type] = 'Vm'
 options[:object_id] = 999999 # some random number

expect { MiqAeEngine.deliver(options) }.to raise_error(ActiveRecord::RecordNotFound)
```

Reply:
@mkanoor sure, thank you for your advice